data-loader/src/index.ts

```diff
@@ -2,27 +2,148 @@ import { parse } from 'yaml';
 import { readFileSync } from 'fs';
 
 import * as z from 'zod';
+import neo4j from 'neo4j-driver'
+
 import { Obj, camelize } from './utils';
+import getDriver from './driver';
 
 const SUBJECTS_FILE_PATH = "./input/subjects.yml";
+const GROUPS_FILE_PATH = "./input/groups.yml";
+const PREREQUISITES_FILE_PATH = "./input/prerequisites.yml";
 
 const subjectSchema = z.object({
   code: z.string(),
   name: z.string(),
   credits: z.number(),
   hasExam: z.boolean(),
   subjectGroup: z.string().nullable(),
-})
+});
 
 const subjectsSchema = z.array(subjectSchema);
 
-async function readSubjects() {
+const groupSchema = z.object({
+  code: z.string(),
+  name: z.string(),
+  minCredits: z.number(),
+});
+
+const groupsSchema = z.array(groupSchema);
+
+const basePrerequisiteSchema = z.object({
+  type: z.enum(['logical', 'subject', 'credits']).optional(),
+  logicalOperator: z.enum(['and', 'or', 'not', 'at_least']).optional(),
+  subjectNeededCode: z.string().optional(),
+  needs: z.enum(['exam', 'course', 'all', 'enrollment']).optional(),
+  subjectCode: z.string().optional(),
+  isExam: z.boolean().optional(),
+}); 
+
+type Prerequisite = z.infer<typeof basePrerequisiteSchema> & {
+  operands?: Prerequisite[];
+};
+
+const prerequisiteSchema: z.ZodType<Prerequisite> = basePrerequisiteSchema.extend({
+  operands: z.lazy(() => prerequisiteSchema.array()).optional(),
+});
+
+const prerequisitesSchema = z.array(prerequisiteSchema);
+
+function readSubjects() {
   const fileBuffer = readFileSync(SUBJECTS_FILE_PATH, 'utf8');
   const objects = Object.values(camelize(parse(fileBuffer) as Obj));
 
-  const subjects = subjectsSchema.parse(objects);
+  return subjectsSchema.parse(objects);
+}
+
+function readGroups() {
+  const fileBuffer = readFileSync(GROUPS_FILE_PATH, 'utf8');
+  const objects = Object.values(camelize(parse(fileBuffer) as Obj));
+
+  return groupsSchema.parse(objects);
+}
+
+function readPrerequisites() {
+  const fileBuffer = readFileSync(PREREQUISITES_FILE_PATH, 'utf8');
+  const objects = Object.values(camelize(parse(fileBuffer) as Obj));
+
+  return prerequisitesSchema.parse(objects);
+}
+
+// Goes over current prerequisite and creates the structure in the database. In is a recursive function.
+function savePrerequisites(prerequisite: Prerequisite, prevId?: string) {
+  // Create Previatura node related to previous or subject with unique id
+  // if logical, always create a new Previatura
```

Comment:
I think this step should be right but I'm worried about the order in which the file is read. Maybe we can discuss it in a meeting.